vividcast/levelclass.lua

```diff
@@ -82,22 +83,22 @@ function level:draw(x,y,rw,rh,sx,sy)
       if current_x ~= new_x or current_y ~= new_y then
         current_x,current_y = new_x,new_y
       end
-    until self:getMapCallback()(current_x,current_y) ~= 0 or
+    until self:getMapCallback()(current_x, current_y) ~= 0 or
       ray_length >= self:getRaycastRange()
 
-    local tile_type = self:getMapCallback()(math.floor(ray_x),math.floor(ray_y))
+    local tile_type = self:getMapCallback()(current_x, current_y)
 
     if ray_length < self:getRaycastRange() then
 
       local tile
-      for i,v in pairs(self:getTiles()) do
+      for _,v in pairs(self:getTiles()) do
         if v.type == tile_type then
           tile = v.tile
         end
       end
       assert(tile,"Tile type `"..tile_type.."` not set.")
 
-      local draw_height = h/ray_length
+      local draw_height = h/(ray_length * math.cos(relative_angle))
```

Comment:
This is the actual change, multiply by the cosine of the relative angle between the player and the line which is calculated [here](https://github.com/josefnpat/VividCast/pull/10/files#diff-26272b8558a685faea5cf72a7a38a8f8R89)